File: 0_1.baekjoon/brute-force/python/BOJ12919.py
from sys import stdin
readline = stdin.readline

S = readline().strip()
T = readline().strip()
result = False
def solve():
    dfs(T[:])
    print(1 if result else 0)

def dfs(string):
    global result
    if len(string) == len(S):
        if string == S:
            result = True
        return

    if string[len(string) - 1] == 'A':
        dfs(string[:len(string) - 1])
    
    if string[0] != 'A':
        dfs(string[::-1][:len(string) - 1])

solve()

File: 0_1.baekjoon/brute-force/java/src/BOJ12919.java
import java.io.*;
import java.util.*;

public class BOJ12919 {
    static String S, T;
    static boolean result = false;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        S = br.readLine();
        T = br.readLine();
        dfs(T);
        System.out.println(result ? 1 : 0);
    }

    public static void dfs(String str) {
        if (str.length() == S.length()) {
            if (str.equals(S))
                result = true;
            return;
        }
        
        int len = str.length();
        
        if (str.charAt(len - 1) == 'A') {
            dfs(str.substring(0, len - 1));
        } 

        if (str.charAt(0) != 'A') {
            dfs(new StringBuilder(str).reverse().substring(0, len - 1));
        }
    }
}